lib/address.ts

```diff
@@ -39,3 +39,12 @@ function xorByte(value: number): number {
 function toPosInt(byte: number): number {
   return byte & 0xff
 }
+
+export const isAddressValid = (address: string) => {
+  if (!address) return false
+
+  const match = address.match(/^[1-9A-HJ-NP-Za-km-z]+$/)
+  const bytes = bs58.decode(address).slice(1)
+
+  return match !== null && match[0] === address && bytes.length >= 32
+}
```

Comment:
This will fail for P2MPKH as @Lbqds said.

Here's a simpler version if we ignore P2MPKH still:

```suggestion
export const isAddressValid = (address = '') =>
  /^[1-9A-HJ-NP-Za-km-z]+$/.test(address) && bs58.decode(address).slice(1).length >= 32
}
```

Reply:
Why will it fail for P2MPKH? It will be larger than 32 afaiu.

Also, I don't like that we default the prop to an invalid address, I'd rather leave it without a default value.

Reply:
Most likely because given a P2MPKH that uses two addresses, it will be larger than say, 34 bytes, but less than X bytes, and anything less or inbetween will be incorrect

The code is equivalent for the default: `if(!address)` is equivalent to `if(address === '')` - `address` is not optional nor is it able to be assigned `undefined`

Reply:
> Most likely because given a P2MPKH that uses say two addresses, it will be larger than say, 34 bytes, but less than X bytes, and anything less or inbetween or more will be incorrect. It's probably not advisable to just use a regexp and length to determine address validity...

This has been confirmed by @Lbqds on Slack:

> the minimum size of P2MPKH is 35 bytes, in most case, the size of P2MPKH is 2 + 32 * n + 1, n is the number of public keys.

Reply:
Adding a default value makes the prop optional. What am I missing? I can call `isAddressValid()`.

Reply:
Ah, TIL default values are still marked optional, makes sense :slightly_smiling_face:  - regardless `isAddressValid()` will still be false. If this is really not preferable then just do `address !== ''`